src/javascript/Switch.js

```diff
@@ -108,23 +114,43 @@ export class Switch {
             this.checkboxBlurred(e);
         }.bind(this), false);
 
-        this.element.addEventListener("click", function(e) {
+        this.element.addEventListener("click", function (e) {
+            if (this.disabled) {
+                e.preventDefault();
+                return
+            }
+
             this.checkboxToggled(e);
         }.bind(this), false);
+
+
+        // Bind changes to the disabled attribute
+        const observer = new MutationObserver((mutations) => {
```

Comment:
NIT: It might be worth changing this to:
```suggestion
        /** @private */
        this.observer = new MutationObserver((mutations) => {
```

While we don't reuse `observer` anywhere right now, it is technically part of the class.